scripts/SE.js

```diff
@@ -428,10 +429,55 @@ SE = {
 		}
 
 		SE.LoadBalances(account, r => {
-			SE.ShowHomeView('balances', { balances: r, account: account }, { a: account });
+			SE.GetScotUserTokens(account, scotTokens => {
+				SE.ShowHomeView('balances', { balances: r, scotTokens: scotTokens, account: account }, { a: account });
+			});
 		});
 	},
 
+	GetScotUserTokens: function(account, callback) {
+		if (!account && SE.User) {
+			account = SE.User.name;
+		}
+
+		$.get(Config.SCOT_API + `@${account}`, { v: new Date().getTime() }, results => {
+			SE.User.ScotTokens = results;
+
+			if (callback) {
+				callback(Object.entries(results));
+			}
+		});
+	},
+
+	ClaimToken: function(symbol) {
+		SE.ShowLoading();
+		
+		const token = SE.Tokens.find(t => t.symbol === symbol);
+		const username = SE.User.name;
+		const amount = SE.User.ScotTokens[symbol].pending_token;
+		const factor = Math.pow(10, token.precision);
+		const calculated = amount / factor;
+
+		const claimData = {
+			symbol
+		};
+
+    if (useKeychain()) {
+      steem_keychain.requestCustomJson(username, 'scot_claim_token', 'Active', JSON.stringify(claimData), `Claim ${calculated} ${symbol.toUpperCase()} Tokens`, function(response) {
```

Comment:
the custom_json can also be sent with the posting authority

Reply:
Is there any disadvantage using one or the other?

Reply:
Using the posting authority is more secure, as the active key is not used, thats all.